tools/c7n_gcp/c7n_gcp/actions/notify.py

```diff
@@ -0,0 +1,153 @@
+             template: policy-template
+             transport:
+               type: pubsub
+               topic: your-notify-topic
+    """
+    batch_size = 250
+
+    schema = {
+        'type': 'object',
+        'anyOf': [
+            {'required': ['type', 'transport', 'to']},
+            {'required': ['type', 'transport', 'to_from']}],
+        'properties': {
+            'type': {'enum': ['notify']},
+            'to': {'type': 'array', 'items': {'type': 'string'}},
+            'owner_absent_contact': {'type': 'array', 'items': {'type': 'string'}},
+            'to_from': ValuesFrom.schema,
+            'cc': {'type': 'array', 'items': {'type': 'string'}},
+            'cc_from': ValuesFrom.schema,
+            'cc_manager': {'type': 'boolean'},
+            'from': {'type': 'string'},
+            'subject': {'type': 'string'},
+            'template': {'type': 'string'},
+            'transport': {
+                'oneOf': [
+                    {'type': 'object',
+                     'required': ['type', 'topic'],
+                     'properties': {
+                         'topic': {'type': 'string'},
+                         'type': {'enum': ['pubsub']},
+                     }}],
+            },
+        }
+    }
+
+    @staticmethod
+    def register_notify_action(registry, _):
+        for resource in registry.keys():
+            klass = registry.get(resource)
+            klass.action_registry.register('notify', Notify)
+
+    def process(self, resources, event=None):
+        self.session = utils.local_session(self.manager.session_factory)
+        project = self.session.get_default_project()
+        message = {
+            'event': event,
+            'account_id': project,
+            'account': project,
+            'region': 'all',
+            'policy': self.manager.data
+        }
+
+        message['action'] = self.expand_variables(message)
+
+        for batch in utils.chunks(resources, self.batch_size):
+            message['resources'] = batch
+            receipt = self.send_data_message(message)
+            self.log.info("sent message:%s policy:%s template:%s count:%s" % (
+                receipt, self.manager.data['name'],
+                self.data.get('template', 'default'), len(batch)))
+
+    def send_data_message(self, message):
+        return self.publish_message(message)
+
+    # Methods to handle GCP Pub Sub topic publishing
+    def publish_message(self, message):
+        """Publish message to a GCP pub/sub topic
+         """
+        topic = self.ensure_topic()
+
+        client = self.session.client('pubsub', 'v1', 'projects.topics')
+
+        try:
+            return client.execute_command('publish', {
+                'topic': topic,
+                'body': {
+                    'messages': {
+                        'data': self.pack(message)
+                    }
+                }
+            })
+        except HttpError as e:
+            if e.resp.status != 404:
```

Comment:
why is this catching 404s?

Reply:
Removed and now just log the error.

Reply:
better not to catch the exception, and let it bubble up